### PR TITLE
Use sockets

### PR DESCRIPTION
**Pull request by jonnyz32**
- Replace db2_publish_message communication to use sockets. Will replace others once we implement the listeners in camel
- Fix the ccsid issue as we aren't using sql anymore
- Fix the performance/scalability issue as socket communication is much faster than sql communication

Thank you to @SanjulaGanepola for helping me to work out some extremely weird ccsid issues which were fixed by adding `TGTCCSID(*JOB)` in the makefile

### CODE REVIEW

File: camel/src/main/java/com/github/theprez/manzan/ManzanMessageFormatter.java [hunk 1]
```diff
@@ -23,7 +23,7 @@ public static String format(final String _in, final Map<String, Object> _mapping
     private final String m_fmtStr;
 
     public ManzanMessageFormatter(final String _fmtStr) {
-        m_fmtStr = _fmtStr;
+        m_fmtStr = _fmtStr.toUpperCase();
```
Comment (SanjulaGanepola) [inline camel/src/main/java/com/github/theprez/manzan/ManzanMessageFormatter.java:26]: Will this change the casing of any text in their specified format? For example, for:
`format=$MESSAGE_ID$ (severity $SEVERITY$): $MESSAGE$`

the words "severity" would be in all caps?

File: camel/src/main/java/com/github/theprez/manzan/routes/event/WatchMsgEvent.java [hunk 2]
```diff
@@ -15,29 +18,31 @@ public class WatchMsgEvent extends ManzanRoute {
     private final String m_schema;
     private final String m_sessionId;
     private final ManzanMessageFormatter m_formatter;
+    private final String m_socketIp = "0.0.0.0";
+    private final String m_socketPort = "8080";
```
Comment (SanjulaGanepola) [inline camel/src/main/java/com/github/theprez/manzan/routes/event/WatchMsgEvent.java:22]: Will we need to make this configurable? If yes, can look into doing this in a separate PR.

File: camel/src/main/java/com/github/theprez/manzan/routes/event/WatchMsgEvent.java [hunk 3]
```diff
@@ -15,29 +18,31 @@ public class WatchMsgEvent extends ManzanRoute {
...
-        .routeId("manzan_msg:"+m_name)
-        .setHeader(EVENT_TYPE, constant(ManzanEventType.WATCH_MSG))
-        .setBody(constant("SeLeCt * fRoM " + m_schema + ".mAnZaNmSg wHeRe SESSION_ID = '"+m_sessionId+"' limit " + m_numToProcess ))
-       // .to("stream:out")
-        .to("jdbc:jt400?outputType=StreamList")
-        .split(body()).streaming().parallelProcessing()
-            .setHeader("id", simple("${body[ORDINAL_POSITION]}"))
-            .setHeader("session_id", simple("${body[SESSION_ID]}"))
+        from(String.format("netty:tcp://%s:%s?sync=false", m_socketIp, m_socketPort))
+            .log("Received raw message: ${body}")
```
Comment (SanjulaGanepola) [inline camel/src/main/java/com/github/theprez/manzan/routes/event/WatchMsgEvent.java:41]: Where does this get logged to?

File: ile/src/SockClient.h [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+    // Method to open a socket and connect to the server
+    bool openSocket(const std::string ip, int port);
+
+    // Method to send a message (string) over the socket
+    bool sendMessage(const std::string message);
+
+    // // Method to send a struct over the socket
+    // template<typename T>
+    // bool sendStruct(const typename T& data);
```
Comment (SanjulaGanepola) [inline ile/src/SockClient.h:19]: I assume this won't be used so it can be removed?

File: ile/src/pub_db2.sqlc [hunk 5]
```diff
@@ -25,6 +32,28 @@ void check_sql_error(int sqlcode, const char* sqlstate)
   }
 }
 
+bool send_message(std::string message){
+    // Create a SocketClient instance
+    SockClient client;
+
+    // Open a socket and connect to server
+    if (!client.openSocket(SERVER, PORT)) {
+        // TODO: How do we want to handle this error? Drop the message, insert into table?
```
Comment (SanjulaGanepola) [inline ile/src/pub_db2.sqlc:41]: Can we open an issue to track this

**Review by SanjulaGanepola — CHANGES_REQUESTED** [review]

File: ile/src/pub_db2.sqlc [hunk 6]
```diff
@@ -33,37 +62,31 @@ void check_sql_error(int sqlcode, const char* sqlstate)
 //     4. implement!
 extern int db2_publish_message(PUBLISH_MESSAGE_FUNCTION_SIGNATURE)
 {
-  exec sql include SQLCA;
```
[the remark this answers is not in the document]
Comment (jonnyz32) [reply ile/src/pub_db2.sqlc:36]: Done

File: ile/src/SockClient.h (hunk 4 above)
Comment (jonnyz32) [reply ile/src/SockClient.h:19]: Yep, removed

File: camel/src/main/java/com/github/theprez/manzan/ManzanMessageFormatter.java (hunk 1 above)
Comment (jonnyz32) [reply camel/src/main/java/com/github/theprez/manzan/ManzanMessageFormatter.java]: Fixed this

File: camel/src/main/java/com/github/theprez/manzan/routes/event/WatchMsgEvent.java (hunk 3 above)
Comment (jonnyz32) [reply camel/src/main/java/com/github/theprez/manzan/routes/event/WatchMsgEvent.java:41]: Yea we can optimize this later

File: test/e2e/msg/snd2q/data.ini [hunk 7]
```diff
@@ -2,4 +2,4 @@
 type=watch
 id=TESTING
 destinations=myfile
-format=$MESSAGE_ID$ (severity $SEVERITY$): $MESSAGE$ 
\ No newline at end of file
+format=$MSGID$ (severity $SEVERITY$): $MESSAGE$
```
[the remark this answers is not in the document]
Comment (jonnyz32) [reply test/e2e/msg/snd2q/data.ini:5]: Fixed this, it's from the construct_json_message

**Review by SanjulaGanepola — APPROVED** [review]
Looks good. Great work

**Pull request closed without merging**